analyses/immune-deconv/01-immune-deconv.R

```diff
@@ -0,0 +1,96 @@
+# Author: Komal S. Rathi
+# Date: 11/11/2019
+# Function: 
+# Script to perform immune characterization using xCell etc.
+
+# load libraries
+suppressPackageStartupMessages(library(optparse))
+suppressPackageStartupMessages(library(tidyverse))
+suppressPackageStartupMessages(library(immunedeconv))
+
+option_list <- list(
+  make_option(c("-p", "--polyaexprs"), type = "character",
+              help = "PolyA Expression data: HUGO symbol x Sample identifiers (.RDS)"),
+  make_option(c("-s", "--strandedexprs"), type = "character",
+              help = "Stranded Expression data: HUGO symbol x Sample identifiers (.RDS)"),
+  make_option(c("-c", "--clin"), type = "character",
+              help = "Clinical file (.TSV)"),
+  make_option(c("-m", "--method"), type = "character",
+              help = "Deconvolution Method"),
+  make_option(c("-b", "--cibersortbin"), type = "character", 
+              help = "Path to Cibersort binary (CIBERSORT.R)"),
+  make_option(c("-g", "--cibersortgenemat"), type = "character", 
+              help = "Path to Cibersort signature matrix (LM22.txt)"),
+  make_option(c("-o","--outputfile"), type = "character",
+              help = "Deconv Output (.RData)")
+)
+
+# Example Run:
+# Rscript analyses/immune-deconv/01-immune-deconv.R \
+# -p 'data/pbta-gene-expression-rsem-fpkm-collapsed.polya.rds' \
+# -s 'data/pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds' \
+# -c 'data/pbta-histologies.tsv' \
+# -m $DECONV_METHOD \
+# -b $CIBERSORT_BIN \
+# -g $CIBERSORT_MAT \
+# -o 'analyses/immune-deconv/results/deconv-output.RData'
+
+# parse parameters
+opt <- parse_args(OptionParser(option_list = option_list))
+polya <- opt$polyaexprs
+stranded <- opt$strandedexprs
+clin.file <- opt$clin
+deconv.method <- opt$method
+cibersort_bin <- opt$cibersortbin 
+cibersort_mat <- opt$cibersortgenemat 
+output.file <- opt$outputfile
+
+print(cibersort_bin)
+print(cibersort_mat)
+# if cibersort_bin and cibersort_mat are defined
+# then, set path to cibersort binary and matrix
+if(cibersort_bin != "NA" & cibersort_mat != "NA"){
+  set_cibersort_binary(cibersort_bin)
+  set_cibersort_mat(cibersort_mat)
+}
+
+# merge expression from polya and stranded data on common genes
+polya <- readRDS(polya)
+stranded <- readRDS(stranded)
+common.genes <- intersect(rownames(polya), rownames(stranded))
+polya <- polya[common.genes,]
+stranded <- stranded[common.genes,]
+expr.input <- cbind(polya, stranded)
```

Comment:
@jaclyn-taroni Quoting a few sentences from two papers that I referred to before merging all the samples together:

```
Comprehensive evaluation of transcriptome-based cell-type quantification methods for immuno-oncology: PMID 31510660
1. xCell uses the variability among the samples for a linear transformation of the output score.
2. xCell does not detect any signal when ran with few non-heterogeneous samples
```

```
xCell: digitally portraying the tissue cellular heterogeneity landscape: PMID 29141660

1. We chose to apply a gene signature enrichment approach over deconvolution methods because of several advantages that the former provides. 
First, gene signatures are rank-based and are therefore suitable for cross-platform transcriptome measurements. 
We showed here that our scores reliably predict enrichment when using different RNA-seq techniques and different microarray platforms. 
They are agnostic to normalization methods or concerns related to batch effects, making them robust to both technical and biological noise.
```